### DISCUSSION TITLE
#71 - cannot import name - from helper function

### DISCUSSION THREAD

**Discussion by Yossi-Hd** [mrdbourke/pytorch-deep-learning#308] [Q&A] [answered]
The helper_functions was downloaded but I get the error:
"ImportError: cannot import name 'plot_predictions' from 'helper_functions' (/content/helper_functions.py)"

I also try to install the helper_function using "!pip install helper_function" but got the same error

**Comment by mrdbourke** [comment, marked as answer]
Hi @Yossi-Hd,

I'm not sure why this may be happening.

But if you're running in Google Colab, you can try the following:

**Step 1**

Run this code in a Google Colab cell:

```
!wget https://raw.githubusercontent.com/mrdbourke/pytorch-deep-learning/main/helper_functions.py
```

**Step 2**

Import `plot_predictions` from `helper_functions.py`. 

```python
from helper_functions import plot_predictions 
```

Running `!pip install helper_functions` won't work as it's not a pip package but rather a Python file on the course GitHub: https://github.com/mrdbourke/pytorch-deep-learning/blob/main/helper_functions.py

## Example  in Google Colab

<img width="1199" alt="Screenshot 2023-02-24 at 5 07 13 pm" src="https://user-images.githubusercontent.com/16750345/221114967-c6001eb0-35f6-48a2-990d-04c6c461767b.png">

**Reply by Yossi-Hd** [reply]
Thank you!
Somehow now when I ran it after a few days it worked

**Comment by nipunikajain** [comment]
HI @mrdbourke 

I am facing the same issue, tried with normal file import and then importing the functions and through get as well. With both the file get's written in colab but import doesn't work.

**Reply by mrdbourke** [reply]
Hi @nipunikajain, 

In this case, I'm not sure what's happening here. 

I can confirm I can start a fresh Google Colab and import the file after downloading it. 

See the demo notebook here: https://colab.research.google.com/drive/1WgEpv0VhI9Wo_dizMMwNuX1AzLtTrDeJ?usp=sharing 

PS if you go and come back from your notebook (and it shuts down), you will have to re-download the file each time.

**Reply by edeltobi** [reply]
I had the same issue. Seems to be a colab specific problem with the invoked runtime, i.e. the contents/helper_functions.py file appears in the file system right away, but is not accessible until the runtime is restarted.

For me the problem was fixed instantly after Runtime -> Restart and run all

Hope this helps
